oceannavigator/frontend/src/components/DatasetSelector.jsx

```diff
@@ -183,6 +200,7 @@ class DatasetSelector extends React.Component {
       newState = {
         variable: newVariable,
         variable_scale: variable.scale,
+        two_dimensional : this.state.datasetVariables.find(v => v.id === newVariable).two_dimensional,
```

Comment:
We've already found the object corresponding to the `newVariable` on line 198. No need to perform the search twice.

```suggestion
        two_dimensional: variable.two_dimensional,
```